source-code/core/src/lint/index.ts

```diff
@@ -0,0 +1,24 @@
+export { lint } from "./linter.js";
+export * from "./query.js";
+export { type Context, parseLintConfigArguments } from "./context.js";
+export {
+  createLintRule,
+  type LintRule,
+  type LintRuleInitializer,
+  type LintRuleId,
+} from "./rule.js";
+export {
+  createLintRuleCollection,
+  type RuleCollectionInitializer,
+} from "./ruleCollection.js";
+
```

Comment:
Do you mean lowering the surface area and waiting for users to demand this feature? 

I would vote yes. Instead of an issue, I would open a discussion though. Issues are tasks that should be worked on soonish.

Reply:
Added the discussions #404, #405 and removed those comments.